internal/client/users.go

```diff
@@ -0,0 +1,89 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/prefecthq/terraform-provider-prefect/internal/api"
+)
+
+var _ = api.UsersClient(&UsersClient{})
+
+type UsersClient struct {
+	hc           *http.Client
+	apiKey       string
+	basicAuthKey string
+	routePrefix  string
+}
+
+// Users is a factory that initializes and returns a UsersClient.
+//
+//nolint:ireturn // required to support PrefectClient mocking
+func (c *Client) Users() (api.UsersClient, error) {
+	return &UsersClient{
+		hc:           c.hc,
+		apiKey:       c.apiKey,
+		basicAuthKey: c.basicAuthKey,
+		routePrefix:  fmt.Sprintf("%s/users", c.endpoint),
+	}, nil
+}
+
+// Read reads a user.
+func (c *UsersClient) Read(ctx context.Context, userID string) (*api.User, error) {
```

Comment:
This is currently failing - I have a feeling you can't hit this endpoint if you aren't that user. I can hit my own user with my own API token, but can't hit Marvin's user.

Reply:
If the token we use in GitHub CI was generated by someone (I'm guessing Eddie?) then we could potentially test against that account.

Alternatively, we could generate an API token as the Marvin user and use that in CI.

Reply:
If this is all correct, we'll want to be clear that you can only use this to manage your personal user - not other users.

Reply:
The Users API does indeed check to ensure "IsSelf", so users can only retrieve/manage information for their own user. Because of this, I created an API key as the Marvin robot user we have in the `github-ci-tests` workspace and added it as a GitHub secret. I also added a second acceptance testing CI step that tests only the `user` resource, providing the different API key and test name.